### PR TITLE
Use ngx_http_realip_module and X-Real-IP.

### PR DESCRIPTION
**Pull request by chappjc**
An option for getting the "real" client IP into the app via the realip module of nginx.

Also update log_format with "$realip_remote_addr".

### CODE REVIEW

File: controllers/main.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ import (
 	"errors"
 	"fmt"
 	"html/template"
+	"net"
```
Comment (jolan) [inline controllers/main.go]: This causes a compile error.

File: controllers/main.go [hunk 2]
```diff
@@ -72,6 +73,31 @@ func randToken() string {
...
+			return parts[0]
+		}
+		return ip
+	}
+
+	realIP := r.Header.Get("X-Real-IP")
+	realIP = getHost(realIP)
+
+	if realIP == "" {
+		log.Infof(`"X-Real-IP" header invalid, using RemoteAddr instead (%s, %v).`,
```
Comment (jolan) [inline controllers/main.go]: Need to remove a format string or add another argument.

Comment (chappjc) [inline controllers/main.go]: No arguments needed, I think.

**Comment by chappjc** [comment]
@jolan OK, done.  :)

**Pull request merged**